File: 339B/main.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using namespace std;

int main() {
    int n, m;
    cin >> n >> m;
    
    vector<int> sp(m);
    for (int i = 0; i < m; i++) {
        cin >> sp[i];
    }
    
    reverse(sp.begin(), sp.end());
    sp.push_back(1);
    reverse(sp.begin(), sp.end());
    
    long long time = 0;
    
    for (int i = 0; i < m; i++) {
        if (sp[i] < sp[i + 1]) {
            time += sp[i + 1] - sp[i];
        } else if (sp[i] > sp[i + 1]) {
            time += n - sp[i] + sp[i + 1];
        }
    }
    
    cout << time << endl;
    
    return 0;
}

File: 339B/main.py
n, m = map(int, input().split())
sp = list(map(int, input().split()))

time = 0

sp.reverse()
sp.append(1)
sp.reverse()

for i in range(m):
    if sp[i] < sp[i + 1]:
        time += sp[i + 1] - sp[i]
    elif sp[i] > sp[i + 1]:
        time += n - sp[i] + sp[i + 1]
    else:
        pass

print(time)